PyStemmusScope/forcing_io.py

```diff
@@ -0,0 +1,167 @@
+import hdf5storage
+import os
+import numpy as np
+import xarray as xr
+
+
+def _calculate_ea(t_air_celcius, rh):
+    """Internal function that calculates the actual vapour pressure (kPa) from the
+    air temperature (degree Celcius) and relative humidity (%)
+
+    Args:
+        t_air_celcius: numpy array containing the air temperature in degrees C.
+
+        rh: numpy array of same shape as t_air_celcius, containing the relative humidity
+            as a percentage (e.g. ranging from 0 - 100).
+
+    Returns:
+        numpy array with the calculated actual vapor pressure
+    """
+    # es = 6.107 * 10**(t_air_celcius*7.5 / (237.3+t_air_celcius))
+    es = 0.61 * np.exp(19.9*t_air_celcius/(273+t_air_celcius))
+    return es * rh/100
+
+
+def _write_matlab_ascii(fname, data, ncols):
+    """Internal function to handle writing data in the Matlab ascii format. Equivalent
+    to `save([-], '-ascii')` in Matlab.
+
+    Args:
+        fname (path or str): Path to the file that should be written
+        data (np.array): Array with data to write to file
+        ncols (int, optional): The number of data columns, required to correctly format
+            the ascii file when writing multiple variables.
+    """
+    matlab_fmt = ' %14.7e'
+    multi_fmt = [matlab_fmt]*ncols
+    multi_fmt[0] = ' ' + multi_fmt[0]
+    np.savetxt(fname, data, multi_fmt)
+
+
+def read_forcing_data(forcing_file):
+    """Reads the forcing data from the provided netCDF file, and applies the required
+    unit conversions before returning the read data.
+
+    Args:
+        forcing_file (str or path): Path to the netCDF file containing the forcing data
+
+    Returns:
+        dict: Dictionary containing the different variables required by STEMMUS_SCOPE
+            for the different forcing files.
+    """
+    ds_forcing = xr.open_dataset(forcing_file)
+
+    # remove the x and y coordinates from the data variables to make the numpy arrays 1D
+    ds_forcing = ds_forcing.squeeze(['x', 'y'])
+
+    data = {}
+    # Expected time format is days (as floating point) since Jan 1st 00:00.
+    data['doy_float'] = (
+        ds_forcing['time'].dt.dayofyear - 1 +
+        ds_forcing['time'].dt.hour/24 +
+        ds_forcing['time'].dt.minute/60/24
+    )
```

Comment:
`datetime.date` has a method `timetuple.tm_yday`. I am wondering if we can use this method somehow. see https://docs.python.org/3/library/datetime.html#datetime.date.timetuple